collect/cnrfc/cnrfc.py

```diff
@@ -496,6 +547,122 @@ def get_ensemble_forecast_watershed(watershed, duration, date_string, acre_feet=
                                  'downloaded': dt.datetime.now().strftime('%Y-%m-%d %H:%M')}}
 
 
+def get_ensemble_forecast_watershed_IO(watershed, duration, date_string, path=None):
+    """
+    from: get_watershed_ensemble_issue_time
+          get_watershed_ensemble_daily
+
+    download seasonal outlook for the watershed as zipped file, unzip, save as csv to path
+
+    Arguments:
+        watershed (str): the forecast group identifier
+        duration (str): forecast data timestep (hourly or daily)
+        date_string (str): the forecast issuance date as a YYYYMMDDHH formatted string
+        path (str): file name including file path/s3 key
+        cnrfc_id (str): the 5-character CNRFC forecast location code
+    Returns:
+        (dict): dictionary with data (dataframe) entry and info metadata dict
+    """
+    units = 'kcfs'
+
+    duration = _validate_duration(duration)
+
+    # store original date_string
+    _date_string = date_string
+
+    # forecast datestamp prefix
+    date_string = _default_date_string(date_string)
+
+    # data source
+    url = 'https://www.cnrfc.noaa.gov/csv/{0}_{1}_hefs_csv_{2}.zip'.format(date_string, watershed, duration)
+
+    # extract CSV from zip object
+    if get_web_status(url):
+        try:
+            csvdata = _get_forecast_csv(url)
+        except zipfile.BadZipFile:
+            print(f'ERROR: forecast for {date_string} has not yet been issued.')
+            raise zipfile.BadZipFile
+    
+    # raise error if user supplied an actual date string but that forecast doesn't exist
+    elif _date_string is not None:
+        print(f'ERROR: forecast for {date_string} has not yet been issued.')
+        raise zipfile.BadZipFile
+
+    # get date/time stamp from ensemble download page
+    time_issued = get_watershed_forecast_issue_time(duration, watershed, date_string)
+
+    # set path for case where path set to None
+    if path is None:
+        path = '{}/{}'.format(os.getcwd(), 
+                                url.split('/')[-1].replace('.zip','.csv'))
+
+    # write csvdata to specified path
+    path = path.replace('/', os.sep)
+    with open(path, 'wb') as f:
+        f.write(csvdata.read())
+
+def get_watershed_forecast(watershed, date_string, forecast_type, duration='daily', path=None):
```

Comment:
If we set duration to None, does it cause any issues?  Since deterministic is an hourly product, it would be a little weird to have a default daily value, even if it doesn't affect the result.

Reply:
If we set the duration to None, the deterministic forecast is returned, no problem. When we try to call the ensemble forecast, we just get an error from the duration check. Not sure if that's the desired behavior.

Reply:
@hyphy8 Since the ensemble forecast needs the user to define a duration, I think that should be the desired behavior.

Reply:
Makes sense. 

Resolved in b2c492e685a258948e88f581b4cd7255b3c71736